autoscheduler/frontend/src/redux/reducers/courseCards.ts

```diff
@@ -63,12 +64,14 @@ export default function courseCards(
         ...state,
         [action.index]: undefined,
       };
-    case UPDATE_COURSE_CARD:
+    case UPDATE_COURSE_CARD: {
+      if (!state[action.index]) return state;
```

Comment:
I assume this change is the actual fix itself? What's your thought process on doing this here vs in the `fetch` like I did for changing mismatched terms?

Reply:
Also why'd you add the brackets around it? I thought you only have to do that if you declare variables

Reply:
> I assume this change is the actual fix itself? What's your thought process on doing this here vs in the `fetch` like I did for checking for mismatched terms?

I think this is better since it makes `addCourseCard` and `updateCourseCard` more distinct: use `addCourseCard` if you're adding a new one, `updateCourseCard` if you're trying to update info of an existing one. I also figured that making the change here would prevent any kinds of similar shenanigans from happening in the future, since we now have to be clear if we really want to add a new course card rather than inadvertently creating one as a side-effect of updating